Creacion atuendo

fbascietto opened the pull request:
Se agregan las funcionalidades para la creación de atuendos.

FernaMera committed:
Creacion de Atuendo de forma Aleatoria

FernaMera committed:
Refactorizacion y Creacion de Atuendos a pedido de Usuario

FernaMera committed:
Agregado metodo para pedir nueva prenda si la previa no va

FernaMera committed:
Agregado metodo para pedir nueva prenda si la previa no va

fbascietto committed:
Marco a Atuendo Builder como transient

fbascietto commented:
Armo un PR porque si no se me complica charlar todo en el aire. 

Actualmente, por lo que entendí de lo que tenemos, los guardarropas van a tener un atuendoBuilder... es medio raro esto, me huele a que estamos pifiando el diseño. Capaz podríamos tener un atuendoSingleton? Solo necesitamos una entidad que se encargue de esto.

fbascietto commented on src/main/java/quemepongoAPI/atuendo/AtuendoBuilder.java:
```diff
@@ -15,8 +18,7 @@ public void crearNuevoAtuendo() {
-    public abstract void buildCabeza(Guardarropa unGuardarropa);
-    public abstract void buildCalzado(Guardarropa unGuardarropa);
+    public abstract void buildPorParte(Guardarropa unGuardarropa, PartesCuerpo unaParte);
+    public abstract void setListaDePartes(List<PartesCuerpo> partes);
```
Siendo la lista de partes algo esencial para el atuendo, no tendría sentido que sea parte del constructor de Atuendo() ?

FernaMera replied:
Y lo es.

fbascietto commented on src/main/java/quemepongoAPI/atuendo/AtuendoBuilderTest.java:
```diff
@@ -5,6 +5,9 @@
+
 import static org.junit.jupiter.api.Assertions.*;
 
 class AtuendoBuilderTest {
```
Perdón por gederla de nuevo con giladas. Los tests los llevaria a todos a una carpeta en el main. O sea, tengamos /src/main/test/atuendo, y ahi tiremos los tests de atuendos. Porque vamos a tener que testear todo y es más facil correrlos todos juntos en este formato.

FernaMera replied:
Si, los puse ahi porque en otro tp no me funcionaba pornerlo en otra carpeta.
Si podes poner todos los tests ennuna sola mejor

FernaMera commented:
Puede ser un singleton, aun asi hay que entregarle el atuendo completo por eso es con un builder. En todo caso para mi (si quieren hacerlo con singleton) seria un builder+singleton

FernaMera committed:
Cambios Varios:
- Test movido a otra carpeta
- Agregado Endpoint para crear atuendo aleatorio
- Agregado Excepcion para atuendos incompletos
- Agregado Excepcion para cuando no se encuentra el guardarropas
- Fix memory leak en cache de prendas pedidas

FernaMera committed:
borrado getters innecesarios

fiorellafontau committed:
Commit de Prueba

Joaco capo

jqnsantosr committed:
Commit test

hola

fiorellafontau committed:
Fio

Chicos, agregue una base en mi compu, complete una conexion y se hace el insert. Lo que no se como hacer es que el dato que enviamos en un POST dese POSTMAN (username e id) tomarlos como parametros y ponerlos en el query. Alguien me da una mano? Luego deberiamos crear la base en un docker o VM.

fiorellafontau committed:
Revert "Fio"

This reverts commit 378763697efbb1ce4c5570f4494b5f0571977396.

fiorellafontau committed:
Cambios de base

Cree la base en mi compu. La conexion funciona al igual que el insert. Lo que no se como hacer es agarrar los parametros desde el POST para el insert. Luego deberiamos crear el docker o vm

FernaMera committed:
Update README.md
